api/envoy/extensions/filters/http/ext_proc/v3/processing_mode.proto

```diff
@@ -35,6 +35,18 @@ message ProcessingMode {
   }
 
   // Control how the request and response bodies are handled
+  // When body mutation by external processor is enabled, ext_proc filter will always remove
+  // the content length header in 3 cases below:
+  // 1) STREAMED BodySendMode
+  // 2) BUFFERED_PARTIAL BodySendMode
+  // 3) BUFFERED BodySendMode + SKIP HeaderSendMode
+  // In these modes, ext_proc filter can not guarantee to set the content length correctly when
+  // body is mutated by external processor later. In http1 codec, removing content length will
+  // enable chunked encoding whenever feasible.
+  // In BUFFERED BodySendMode + SEND HeaderSendMode, content length header is allowed but it is
```

Comment:
This all makes sense as previously discussed offline. I do think this is a bit abbreviated and requires a fair bit of detailed HTTP knowledge. This is effectively public customer facing docs for folks who use ext_proc in products (cough), so I'd recommend more verbose than less.

Reply:
Updated. PTAL